src/fhirtypes/common.ts

```diff
@@ -1601,3 +1601,40 @@ export function findImposeProfiles(sd: any): string[] | undefined {
     return imposeProfiles;
   }
 }
+
+// Used to check if the entity used in Canonical() references a contained resource.
+// Checks a list of validated rules at any contained path for a matching value from the
+// Canonical() keyword and returns the matching resource's id.
+export function getMatchingContainedReferenceId(
+  value: string,
+  containedResources: { pathParts: PathPart[]; assignedValue: any }[]
+) {
+  const matchingContainedResource = containedResources.find(
+    r =>
+      r.assignedValue?.url === value ||
+      r.assignedValue?.name === value ||
+      r.assignedValue?.id === value
+  );
+  const matchingContainedResourceId = containedResources.find(
+    r => r.pathParts.slice(-1)[0].base === 'id'
+  );
+  const hasMatchingContainedResourceId = containedResources.some(
+    r => r.pathParts.slice(-1)[0].base === 'id' && r.assignedValue === value
+  );
+  const hasMatchingContainedResourceName = containedResources.some(
+    r => r.pathParts.slice(-1)[0].base === 'name' && r.assignedValue === value
+  );
+  const hasMatchingContainedResourceUrl = containedResources.some(
+    r => r.pathParts.slice(-1)[0].base === 'url' && r.assignedValue === value
+  );
+  if (matchingContainedResource != null) {
+    return matchingContainedResource.assignedValue.id;
+  } else if (
```

Comment:
Since the return value is known right away if `matchingContainedResource` is not null, this check could happen right after trying to find `matchingContainedResource`.

Reply:
Fixed in [e4837bd](https://github.com/FHIR/sushi/pull/1405/commits/e4837bd810dd83692f6be497b890f27510ceb6e6)